Compiler/src/typeinfer.jl

```diff
@@ -181,6 +186,10 @@ function finish!(interp::AbstractInterpreter, caller::InferenceState, validation
     return nothing
 end
 
+function cache_result!(interp::AbstractInterpreter, mi::MethodInstance, ci::CodeInstance)
+    code_cache(interp)[mi] = ci
+end
```

Comment:
```suggestion
function cache_result!(interp::AbstractInterpreter, result::InferenceResult, ci::CodeInstance)
    code_cache(interp)[result.linfo] = ci
end
```

We should probably keep the `cache_result!` signature to include `result::InferenceResult`, since if an external `AbstractInterpreter` implements its own cache, it will need the `result.valid_worlds` information.
(although most of them are currently using `Compiler.InternalCodeCache` anyway)